src/services/route/get-all.ts

```diff
@@ -1,25 +1,44 @@
-import { asc, eq, sql } from "drizzle-orm"
+import { and, asc, eq, gte, sql } from "drizzle-orm"
 import { InternalServerError } from "elysia"
 import Cache from "../../commons/utils/cache"
 import { handleError } from "../../commons/utils/error"
 import { logger } from "../../commons/utils/log"
 import { db, dbSchema } from "../../db"
-import { Schedule } from "../../db/schema"
+import { Schedule, Station } from "../../db/schema"
 import { getSecondsRemainingFromNow } from "../../commons/utils/date"
 
 export const getAll = async (trainId: string) => {
   try {
-    const cache = new Cache<Schedule[]>(`route-${trainId}`, {
-      ttl: getSecondsRemainingFromNow(),
-    })
+    const cache = new Cache<(Schedule & { stationName: Station["name"] })[]>(
+      `route-${trainId}`,
+      {
+        ttl: getSecondsRemainingFromNow(),
+      },
+    )
 
     const cached = await cache.get()
 
     if (cached) return cached
 
-    const schedules = await db.query.schedule.findMany({
-      where: eq(dbSchema.schedule.trainId, trainId),
-      orderBy: [asc(dbSchema.schedule.timeEstimated)],
+    const result = await db
+      .select()
+      .from(dbSchema.schedule)
+      .leftJoin(
+        dbSchema.station,
+        eq(dbSchema.schedule.stationId, dbSchema.station.id),
+      )
+      .where(eq(dbSchema.schedule.trainId, trainId))
+      .orderBy(asc(dbSchema.schedule.timeEstimated))
+
+    const schedules = result.map((res) => ({
+      ...res.schedule,
+      stationName: res.station?.name || null,
+    }))
+    // Add the last station schedule
+    schedules.push({
+      ...schedules[0],
+      stationName: schedules[0].destination,
+      timeEstimated: schedules[0].destinationTime,
```

Comment:
this one is incomplete, since we should also populate the stationId with the correct ID. 

Maybe we need to query it first, or update the db schema @abielzulio